holoviews/plotting/bokeh/callbacks.py

```diff
@@ -1,446 +1,225 @@
+import json
 from collections import defaultdict
 
-import numpy as np
 import param
+import numpy as np
+from bokeh.models import CustomJS
 
-from ...core.data import ArrayColumns, PandasInterface
-from .util import compute_static_patch, models_to_json
-
-from bokeh.models import CustomJS, TapTool, ColumnDataSource
-from bokeh.core.json_encoder import serialize_json
-from bokeh.io import _CommsHandle
-from bokeh.util.notebook import get_comms
+from ...streams import (Stream, PositionXY, RangeXY, Selection1D, RangeX,
+                        RangeY, PositionX, PositionY, Bounds)
+from ..comms import JupyterCommJS
 
 
-class Callback(param.ParameterizedFunction):
+def attributes_js(attributes):
     """
-    Callback functions provide an easy way to interactively modify
-    the plot based on changes to the bokeh axis ranges, data sources,
-    tools or widgets.
-
-    The data sent to the Python callback from javascript is defined
-    by the plot attributes and callback_obj attributes and optionally
-    custom javascript code.
-
-    The user should define any plot_attributes and cb_attributes
-    he wants access to, which will be supplied in the form of a
-    dictionary to the call method. The call method can then apply
-    any processing depending on the callback data and return the
-    modified bokeh plot objects.
+    Generates JS code to look up attributes on JS objects from
+    an attributes specification dictionary.
+    """
+    code = ''
+    for key, attr_path in attributes.items():
+        data_assign = "data['{key}'] = ".format(key=key)
+        attrs = attr_path.split('.')
+        obj_name = attrs[0]
+        attr_getters = ''.join(["['{attr}']".format(attr=attr)
+                                for attr in attrs[1:]])
+        code += ''.join([data_assign, obj_name, attr_getters, ';\n'])
+    return code
+
+
+class Callback(param.Parameterized):
+    """
+    Provides a baseclass to define callbacks, which return data from
+    bokeh models such as the plot ranges or various tools. The callback
+    then makes this data available to any streams attached to it.
+
+    The defintion of a callback consists of a number of components:
+
+    * handles    :  The handles define which object the callback will be
+                    attached on.
+
```

Comment:
What kind of object? HoloViews objects? If so, it should say so. Also, 'definition' is misspelt above...